Initial import of "Protected API calling Protected API OBO" scenario (#5)
Azure-Samples/ms-identity-docs-code-java: access-protected-api-obo/README.md

<!-- Keeping yaml frontmatter commented out for now
---
# Metadata required by https://docs.microsoft.com/samples/browse/
# Metadata properties: https://review.docs.microsoft.com/help/contribute/samples/process/onboarding?branch=main#add-metadata-to-readme
languages:
- java
page_type: sample
name: "Java API written in Spring Boot that both protects its own endpoints and calls Graph on-behalf-of the client."

> description: "This Java API that protects it's own endpoints using JWT scope validation and calls Microsoft Graph on behalf of the client. The code in this sample is used by one or more articles on docs.microsoft.com."

Reviewer: it's should be its

Author: Thanks, addressed in https://github.com/Azure-Samples/ms-identity-docs-code-java/pull/5/commits/1c104010698da35ee19a6b4ad528de2ba26df54b

Author: Marking resolved (@skabou doesn't have permission to do so)

> description: "This Java API that protects it's own endpoints using JWT scope validation and calls Microsoft Graph on behalf of the client. The code in this sample is used by one or more articles on docs.microsoft.com."

Author: ```suggestion
description: "This Java API that protects its own endpoints using JWT scope validation and calls Microsoft Graph on behalf of the client. The code in this sample is used by one or more articles on docs.microsoft.com."
```

<!-- Keeping yaml frontmatter commented out for now
---
# Metadata required by https://docs.microsoft.com/samples/browse/
# Metadata properties: https://review.docs.microsoft.com/help/contribute/samples/process/onboarding?branch=main#add-metadata-to-readme
languages:
- java
page_type: sample

> name: "Java API written in Spring Boot that both protects its own endpoints and calls Graph on-behalf-of the client."

Reviewer 2: ```suggestion
name: "Java Spring Boot web API that protects its own endpoints and calls Microsoft Graph on behalf of the client application."
```

description: "This Java API that protects its own endpoints using JWT scope validation and calls Microsoft Graph on behalf of the client. The code in this sample is used by one or more articles on docs.microsoft.com."
products:
- azure
- azure-active-directory
- ms-graph

> urlFragment: ms-identity-docs-code-webapp-java

Reviewer 2: ```suggestion
urlFragment: ms-identity-docs-code-webapi-java

```

---
-->

> # Java Spring Boot | web API | access control (protected routes), protected web API access (Microsoft Graph)  | Microsoft identity platform

Reviewer 2: ```suggestion
# Java Spring Boot | web API | access control (protected routes), protected web API access (Microsoft Graph) | Microsoft identity platform
```

<!-- Build badges here
![Build passing.](https://img.shields.io/badge/build-passing-brightgreen.svg) ![Code coverage.](https://img.shields.io/badge/coverage-100%25-brightgreen.svg) ![License.](https://img.shields.io/badge/license-MIT-green.svg)
-->

> This Java Spring Boot web API uses the Microsoft Authentication Library (MSAL) for Java to require authorization to access a JWT-protected route, which then calls Microsoft Graph on behalf of the client.

Reviewer 2: ```suggestion
This Java Spring Boot web API uses the Microsoft Authentication Library (MSAL) for Java to require authorization for clients accessing a JWT-protected route, and then calls Microsoft Graph on behalf of the client.
```

```console
$ curl http://localhost:8080/me -H "Authorization: Bearer {valid-access-token}"
{
  "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#users/$entity",
  "businessPhones": ["+1 (999) 5550001"],

>   "displayName": "Hans Gruber",

Reviewer 2: Per CELA, we have to use _our_ fictitious names for organizations and people. Let's use **Contoso** for the org, and **Contoso Employee** for now.

The username I used to use is no longer usable (this is a good story, remind me on Monday to tell you about it.).

Author: Updated to "Contoso Employee" in both spots in this README.

Changed in https://github.com/Azure-Samples/ms-identity-docs-code-java/pull/5/commits/2738ee798b1c4ad5106dbd3642a9b6dff6075020

| **Supported account types**    | **Accounts in this organizational directory only (Single tenant)**   | Suggested value for this sample.                                                 |
| **Platform type**              | _None_                                                               | No redirect URI required; don't select a platform.                               |
| **Scopes defined by this API** | **Scope name**: `user_impersonation`<br/>**Who can consent?**: **Admins and users**<br/>**Admin consent display name**: `Act on behalf of the user`<br/>**Admin consent description**: `Allows the API to act on behalf of the user.`<br/>**User consent display name**: `Act on your behalf`<br/>**User consent description**: `Allows the API to act on your behalf.`<br/>**State**: **Enabled** | Required scope for this sample. |
| **

> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.

### 2. Update code sample with app registration values

> Open the [_application.yml_](src/main/resources/application.yml) file and modify the four Azure Active Directory configuration properties using the values from your [app's registration in the Azure portal](https://docs.microsoft.com/azure/active-directory/develop/quickstart-configure-app-expose-web-apis).

Reviewer 2: ```suggestion
In [_application.yml_](src/main/resources/application.yml), update the following Azure AD configuration properties with the values from your [app's registration in the Azure portal](https://docs.microsoft.com/azure/active-directory/develop/quickstart-configure-app-expose-web-apis).
```
